spring-web-modules/spring-session/spring-session-redis/src/main/java/com/baeldung/spring/session/SecurityConfig.java

```diff
@@ -25,13 +28,12 @@ public InMemoryUserDetailsManager userDetailsService(PasswordEncoder passwordEnc
 
     @Bean
     public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
-        http.httpBasic()
-            .and()
-            .authorizeRequests()
-            .antMatchers("/")
-            .hasRole("ADMIN")
-            .anyRequest()
-            .authenticated();
```

Comment:
Hi @Bipinkumar27 

Could you analyze:
 .sessionManagement(httpSecuritySessionManagementConfigurer -> httpSecuritySessionManagementConfigurer.sessionCreationPolicy(SessionCreationPolicy.ALWAYS))
 
 Have you tested to check this works as before?

Reply:
Yes @anastasiosioannidis,

In Spring Security 6, the SecurityContextPersistenceFilter and SessionManagementFilter are not set by default.
It required in our use case of storing the session data in Redis in SessionControlIntegrationTest Inside testRedisControlsSession() method 

After doing the above changes it got fixed and test executed successfully

Reply:
However, currently I have set the session creation policy to AWAYS which we can change to IF_REQUIRED.

httpSecuritySessionManagementConfigurer.sessionCreationPolicy(SessionCreationPolicy.ALWAYS) 

Let me know if that needs to change.

Reply:
Hi @Bipinkumar27 

Since before it was if-required, I guess we could revert to this.
Please check if tests work.

Reply:
Also check if we can remove the lambda and simplify it

Reply:
Hi @anastasiosioannidis 
As highlighted earlier, it's imperative to ensure explicit configuration of session management post-transition to Spring 6, as adopted in parent-boot-3. This became necessary after a test case broke due to improper session management configuration.

Regarding simplification through lambda expressions, although it may lead to larger method sizes, it's noteworthy that lambda expressions are widely utilized in Spring documentation. Hence, it would be beneficial for us to adopt the same practice, aligning with Spring standards.

Please share your views

Reply:
Hi @Bipinkumar27 

Thanks for the analysis.
I get build/test errors for the integration profile (Redis couldn't start). Could you share how you ran it?

Reply:
Hey @anastasiosioannidis,

I have run it by using the below cmd:

`mvn clean install -U -Pintegration-jdk9-and-above --pl spring-web-modules/spring-session/spring-session-redis`

<img width="1792" alt="Screenshot 2024-04-05 at 7 24 49 PM" src="https://github.com/eugenp/tutorials/assets/78090485/b0fcb426-0783-406d-8a6a-fc5f8d94075e">



Also, I have run this on Inteij IDE :
<img width="1792" alt="Screenshot 2024-04-05 at 7 27 01 PM" src="https://github.com/eugenp/tutorials/assets/78090485/f51beb6b-59e2-42f7-baea-21b5f8cde3d0">


I am not sure why Redis is not running on your local machine may be due to Redis default port not might occupied by other task.